plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/NewApplicationHandler.java

```diff
@@ -31,10 +34,15 @@
 
 	@Override
 	public Object execute(ExecutionEvent event) throws ExecutionException {
-		IProject project = UIUtils.getFirstElement(HandlerUtil.getCurrentSelection(event), IProject.class);
-		NewApplicationWizardModel model = new NewApplicationWizardModel(project);
-		NewApplicationWizard wizard = new NewApplicationWizard(model);
-		WizardUtils.openWizard(wizard, HandlerUtil.getActiveShell(event));
+		try {
+			IProject project = UIUtils.getFirstElement(HandlerUtil.getCurrentSelection(event), IProject.class);
+			Connection connection = ConnectionsRegistryUtil.getConnectionFor(project);
+			NewApplicationWizardModel model = new NewApplicationWizardModel(project, connection.getResourceFactory());
+			NewApplicationWizard wizard = new NewApplicationWizard(model);
+			WizardUtils.openWizard(wizard, HandlerUtil.getActiveShell(event));
+		}catch(NullPointerException e) {
```

Comment:
@adietish We should never have a null pointer in this situation because connection should never be null; it could only be null if we stubbed it and didnt retrieve it from a connection.  Maybe we should consider making ConnectionsRegistryUtil.getConnectionFor(project) throw an error or take an argument for it to throw an error which would make it more explicit as to whats going on.